File: apps/golang/face_pad.go
package main

import (
	"fmt"
	"oneml/face"
	"os"
	"path"
)

func main() {
	currentPath, _ := os.Getwd()
	testAssetPath := path.Join(currentPath, "../../assets/images")

	// Initialize oneML modules
	licenseManager := face.NewLicenseManager()
	licenseManager.ActivateTrial()

	oneMLUtils := face.NewUtils(licenseManager)
	padRgb := face.NewFacePad(face.PadType_Rgb, licenseManager)
	padPaper := face.NewFacePad(face.PadType_Paper, licenseManager)
	defer face.DeleteUtils(oneMLUtils)
	defer face.DeleteFacePad(padPaper)
	defer face.DeleteFacePad(padRgb)

	// PAD RGB
	// load Image
	filePath := path.Join(testAssetPath, "pad-rgb-set/spoof/1.jpg")
	input := oneMLUtils.ReadImageCV(filePath)

	result := padRgb.Classify(input)

	fmt.Println("Status: ", result.GetReturnStatus())
	fmt.Printf("Spoof probability: %.6f\n", result.GetSpoofProb())
	fmt.Println("Spoof classification:", result.IsSpoof())

	// PAD PAPER
	// load Image
	filePath = path.Join(testAssetPath, "pad-paper-set/spoof/1.jpg")
	input = oneMLUtils.ReadImageCV(filePath)

	result = padPaper.Classify(input)

	fmt.Println("Status: ", result.GetReturnStatus())
	fmt.Printf("Spoof probability: %.6f\n", result.GetSpoofProb())
	fmt.Println("Spoof classification:", result.IsSpoof())

}

File: apps/python/face_pad.py
import pathlib
import os

from oneML import faceAPI as api

os.environ['ONEML_CPP_MIN_LOG_LEVEL'] = "ERROR"
assets_path = os.path.join(
    str(pathlib.Path(__file__).resolve().parent.parent.parent.absolute()),
    "assets/images",
)


def main():
    manager = api.LicenseManager()
    manager.activate_trial()
    pad_rgb = api.FacePad(api.PadType_Rgb, manager)
    pad_paper = api.FacePad(api.PadType_Paper, manager)
    utils = api.Utils(manager)

    # PAD RGB
    img = utils.read_image_cv(
        os.path.join(assets_path, "pad-rgb-set/spoof/1.jpg")
    )

    result = pad_rgb.classify(img)

    print("Status: " + str(result.get_return_status()))
    print("Spoof probability: " + "".join("{:.6f}".format(result.get_spoof_prob())))
    print("Spoof classification: " + str(result.is_spoof()))

    # PAD PAPER
    img = utils.read_image_cv(
        os.path.join(assets_path, "pad-paper-set/spoof/1.jpg")
    )

    result = pad_paper.classify(img)

    print("Status: " + str(result.get_return_status()))
    print("Spoof probability: " + "".join("{:.6f}".format(result.get_spoof_prob())))
    print("Spoof classification: " + str(result.is_spoof()))


if __name__ == "__main__":
    main()